delta/testkit/src/main/scala/ch/epfl/bluebrain/nexus/testkit/mu/ce/CatsEffectAssertions.scala

```diff
@@ -189,6 +194,24 @@ trait CatsEffectMUnitAssertions { self: Assertions =>
     def intercept[T <: Throwable](expectedError: T)(implicit T: ClassTag[T], loc: Location): IO[T] =
       interceptErrorIO[T](expectedError)(io)
 
+    def eventually(expected: A, retryWhen: Throwable => Boolean)(implicit patience: PatienceConfig): IO[Unit] = {
+      val strategy = RetryStrategy[Throwable](
+        MaximumCumulativeDelayConfig(patience.timeout, patience.interval),
+        retryWhen,
+        onError = (_, _) => UIO.unit
+      )
+      assertEquals(expected, patience.timeout)
+        .retry(strategy)
+    }
```

Comment:
I wonder if it's more flexible to pass an assertion here. 

Perhaps in order to keep it both flexible and consistent: 

- This could be renamed `eventuallyEquals`
- A "generic" assertion could be called `eventually`

I wonder if actually you could achieve syntax like this:

`doAnOperation.assertEquals(1).eventually` ? With an implicit class taking `IO[Assertion]`

Of course this can be done another day but it's interesting

Reply:
Maybe, I did not want to modify that now as it will make it harder to migrate the other tests depending on it.
And we will to go back at it when we migrate the retry strategy (there is still a `UIO.unit` here)